account_invoice_report_by_partner/models/account.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def invoice_print(self):
+        result = super(AccountInvoice, self).invoice_print()
```

Comment:
I wouldn't call super() here, just postpone it when we know the partner_id doesn't have an invoice_report_id associated to it so within "if not action_name". That's because otherwise we might call twice the .get_action() when partner_id as a report associated to it.